{"cells":[{"cell_type":"code","execution_count":null,"metadata":{"id":"wsXuKtF3uuGe"},"outputs":[],"source":["!pip install --upgrade streamlit\n","!pip install pyngrok\n","!pip install spacy -U\n","!pip install spacy-transformers\n","!pip install pytesseract -U\n","!apt install tesseract-ocr\n","!apt install libtesseract-dev"]},{"cell_type":"code","execution_count":null,"metadata":{"colab":{"background_save":true,"base_uri":"https://localhost:8080/"},"id":"SLqzA3pNu7O4"},"outputs":[{"name":"stdout","output_type":"stream","text":["NgrokTunnel: \"https://31f7-34-125-173-70.ngrok-free.app\" -\u003e \"http://localhost:80\"\n","\n","Collecting usage statistics. To deactivate, set browser.gatherUsageStats to False.\n","\u001b[0m\n","\u001b[0m\n","\u001b[34m\u001b[1m  You can now view your Streamlit app in your browser.\u001b[0m\n","\u001b[0m\n","\u001b[34m  Network URL: \u001b[0m\u001b[1mhttp://172.28.0.12:80\u001b[0m\n","\u001b[34m  External URL: \u001b[0m\u001b[1mhttp://34.125.173.70:80\u001b[0m\n","\u001b[0m\n","2024-02-05 11:48:30.096 Uncaught app exception\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/elements/image.py\", line 375, in image_to_url\n","    with open(image, \"rb\") as f:\n","FileNotFoundError: [Errno 2] No such file or directory: '/content/drive/MyDrive/4th_Year/DSC4996/janashakthi group (jxg).png'\n","\n","During handling of the above exception, another exception occurred:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/runtime/memory_media_file_storage.py\", line 165, in _read_file\n","    with open(filename, \"rb\") as f:\n","FileNotFoundError: [Errno 2] No such file or directory: '/content/drive/MyDrive/4th_Year/DSC4996/janashakthi group (jxg).png'\n","\n","The above exception was the direct cause of the following exception:\n","\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 535, in _run_script\n","    exec(code, module.__dict__)\n","  File \"/content/drive/MyDrive/4th_Year/DSC4996/app.py\", line 55, in \u003cmodule\u003e\n","    col.image(\"/content/drive/MyDrive/4th_Year/DSC4996/janashakthi group (jxg).png\", use_column_width='auto')\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/runtime/metrics_util.py\", line 397, in wrapped_func\n","    result = non_optional_func(*args, **kwargs)\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/elements/image.py\", line 165, in image\n","    marshall_images(\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/elements/image.py\", line 534, in marshall_images\n","    proto_img.url = image_to_url(\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/elements/image.py\", line 385, in image_to_url\n","    url = runtime.get_instance().media_file_mgr.add(image, mimetype, image_id)\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/runtime/media_file_manager.py\", line 222, in add\n","    file_id = self._storage.load_and_get_id(\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/runtime/memory_media_file_storage.py\", line 116, in load_and_get_id\n","    file_data = self._read_file(path_or_data)\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/runtime/memory_media_file_storage.py\", line 168, in _read_file\n","    raise MediaFileStorageError(f\"Error opening '{filename}'\") from ex\n","streamlit.runtime.media_file_storage.MediaFileStorageError: Error opening '/content/drive/MyDrive/4th_Year/DSC4996/janashakthi group (jxg).png'\n","2024-02-05 11:54:45.461 Uncaught app exception\n","Traceback (most recent call last):\n","  File \"/usr/local/lib/python3.10/dist-packages/streamlit/runtime/scriptrunner/script_runner.py\", line 535, in _run_script\n","    exec(code, module.__dict__)\n","  File \"/content/drive/MyDrive/4th_Year/DSC4996/app.py\", line 99, in \u003cmodule\u003e\n","    Amount = int(entity.text)\n","ValueError: invalid literal for int() with base 10: 'specialist'\n"]}],"source":["from pyngrok import ngrok\n","# !ngrok config add-authtoken 2OQr3VN0TkWlEyAlKPB64KMsgH1_itXF5MuXsTKsWa8iDgDy\n","ngrok.set_auth_token(\"2bEAjl5Ih4N4PmxpGIHYWrZjjd0_4teLZaUqZcsVbheoseeNu\")\n","url = ngrok.connect()\n","print(url)\n","!streamlit run --server.port 80 /content/drive/MyDrive/4th_Year/DSC4996/app.py"]},{"cell_type":"code","execution_count":2,"metadata":{"executionInfo":{"elapsed":522,"status":"ok","timestamp":1707126305467,"user":{"displayName":"Hussain Hisham","userId":"02921849162064616923"},"user_tz":-330},"id":"AWJ9GdjqvZM4"},"outputs":[],"source":["ngrok.kill()"]}],"metadata":{"accelerator":"GPU","colab":{"authorship_tag":"ABX9TyNHVOqu0ATqDIoLNl4P6fF0","mount_file_id":"1ykuPo9Y3hpsxchZQ6xAtDRiLMUPTii0c","name":"","version":""},"kernelspec":{"display_name":"Python 3","name":"python3"},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}